src/relayserver/RelayServer.ts

```diff
@@ -299,12 +307,47 @@ export class RelayServer extends EventEmitter {
     process.exit(1)
   }
 
+  trustedPaymastersGasLimits: {[key: string]: PaymasterGasLimits} = {}
+
+  /***
+   * initialize data from trusted paymasters.
+   * "Trusted" paymasters means that:
+   * - we trust their code not to alter the gas limits (getGasLimits returns constants)
+   * - we trust preRelayedCall to be consistent: off-chain call and on-chain calls should either both succeed
+   *    or both revert.
+   * - given that, we agree to give the requested acceptanceBudget (since breaking one of the above two "invariants"
+   *    is the only cases where the relayer will have to pay for this budget)
+   *
+   * @param paymasters list of trusted paymaster addresses
+   */
+  async _initTrustedPaymasters (paymasters: string[] = []): Promise<void> {
+    this.trustedPaymastersGasLimits = {}
+    for (const paymasterAddress of paymasters) {
+      const paymaster = await this.contractInteractor._createPaymaster(paymasterAddress)
+      const gasLimits = await paymaster.getGasLimits().catch(e => {
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+        throw new Error(`not a valid paymaster address in trustedPaymasters list: ${paymasterAddress}: ${e.message}`)
+      })
+      this.trustedPaymastersGasLimits[paymasterAddress.toLowerCase()] = gasLimits
+    }
+  }
+
+  _getPaymasterMaxAcceptanceBudget (paymaster?: string): IntString {
+    const limits = this.trustedPaymastersGasLimits[paymaster?.toLocaleLowerCase() ?? '']
```

Comment:
You don't need to check `this.trustedPaymastersGasLimits['']` IMO. Just return `maxAcceptanceBudget`

Reply:
I do, since I want to return the default either if no paymaster is given, or if a wrong one is given...